### ISSUE TITLE
Documentation for using the pre-trained model?

### ISSUE THREAD

**Issue by manish-kumar-iisc** [BUTSpeechFIT/VBx#44]
what does out_labels represents?
How can pretrained model used to get diarization outputs? Is there any doucmentation,
Model is executing well on the example given in repo. 
How to generated rttm files?

**Comment by fnlandini** [comment]
Hi,
Unfortunately, there is no written documentation. However, the example script (or the other recipes) are somewhat self-explanatory once you understand how the algorithm works (explained in the papers we reference). I recommend you to look at the parameters passed in the scripts to see how to use the code. 
Anyway, given the other issue you opened, I think you managed to run the code. If so, and you have no other questions regarding how to run the code, please let me know and we can close this issue.

**Comment by manish-kumar-iisc** [comment]
Thanks for help, I am able to run the model.

- Have you compared your pretrained model performance, with pyannote?, I have checked you in result in DIHARD-3 your model DER turned out way better than pyannote, but i guess you first trined on DIHARD -3 dataset & then tested result on test set. I was curious how is your pretrained model performance, have you have compared it with other models?


- I am using [this](https://pytorch.org/hub/snakers4_silero-vad_vad/) model for voice activity detection, which model you have used to generate .lab file or voice acitvity detection?

**Comment by fnlandini** [comment]
I have not compared the results with pyannote. I see that pyannote with default parameters should reach [22.2](https://github.com/pyannote/pyannote-audio#benchmark). If you ran our [DIHARD2 recipe](https://github.com/BUTSpeechFIT/VBx/blob/master/DIHARD2_run.sh), you should probably have obtained better results. The hyperparameters of that recipe were tuned on the dev set of DIHARD 2 which is very similar to the dev set of DIHARD 3. You could find some of our results with VBx in our [system description](https://dihardchallenge.github.io/dihard3/system_descriptions/dihard3_system_description_team55.pdf) for the challenge. Those systems are not exactly like the one in the repository but very similar.
The [winning](https://dihardchallenge.github.io/dihard3/system_descriptions/dihard3_system_description_team54.pdf) and [runner up](https://dihardchallenge.github.io/dihard3/system_descriptions/dihard3_system_description_team115.pdf) teams of DIHARD 3 challenge used VBx as part of their systems as well and you have some extra comparisons in their system descriptions. You will see that VBx is not the best model nowadays but it provides very competitive results. You can find some other comparisons [here](https://reader.elsevier.com/reader/sd/pii/S0885230821000619?token=94F0B1699F1004753F8D9D14B03DE2145A4AA5644B3960DF1D0368FDEDB7863A9BFFE41531FFB2D0A7E614AE1F2D788C&originRegion=eu-west-1&originCreation=20220606101635) where there are a few other systems for AMI (including pyannote) and Callhome.

Regarding the VAD system, I have not used that particular model, mainly because pyannote should work better (see [Table 1](https://arxiv.org/pdf/2104.04045.pdf)). I have recently used pyannote's VAD or [this one](http://kaldi-asr.org/models/m4) and they do a reasonable job. You can also find [here](https://github.com/BUTSpeechFIT/VBx/blob/v1.1_VoxConverse2020/VAD/energy_VAD.py) a simpler model that we released some time ago.

**Comment by manish-kumar-iisc** [comment]
This is super useful information. Thanks for that much.
I will try diarization models, VBx & others & see how it turns out.
Let me reach out you, in case of any other query. Then we can close this issue.

**Comment by fnlandini** [comment]
Closing due to inactivity. Feel free to reopen if you see fit.
